src/test/scala/com/amazon/deequ/comparison/ReferentialIntegrityTest.scala

```diff
@@ -474,4 +558,123 @@ class ReferentialIntegrityTest extends AnyWordSpec with SparkContextSpec {
       assert(result.isInstanceOf[ComparisonSucceeded])
     }
   }
+
+  "Referential Integrity Row Level Test" should {
+    "works for multiple columns" in withSparkSession { spark =>
+      import spark.implicits._
+
+      val rdd1 = spark.sparkContext.parallelize(Seq(
+        (1, "New York", "NY"),
+        (2, "Wisconsin", "WI"),
+        (3, "Texas", "TX"),
+        (4, "Canada", "CA"))) // Incorrect row
+      val ds1 = rdd1.toDF("id", "state name", "state")
+
+      val rdd2 = spark.sparkContext.parallelize(Seq(
+        (1, "New York", "NY"),
+        (2, "Wisconsin", "WI"),
+        (3, "Texas", "TX"),
+        (4, "California", "CA"))) // Reference has correct row
+      val ds2 = rdd2.toDF("id", "state name", "state")
+
+      val cols = Seq("state name", "state")
+      val outcomeCol = "row_level_outcome"
```

Comment:
Can you add an assertion for the columns returned? It should be exactly `("id", "state name", "state", "row_level_outcome")`

Reply:
Done.